pkg/api/v1beta3/types.go

```diff
@@ -934,7 +934,7 @@ type ServiceSpec struct {
 	Ports []ServicePort `json:"ports" description:"ports exposed by the service"`
 
 	// This service will route traffic to pods having labels matching this selector. If null, no endpoints will be automatically created. If empty, all pods will be selected.
-	Selector map[string]string `json:"selector" description:"label keys and values that must match in order to receive traffic for this service; if empty, all pods are selected, if not specified, endpoints must be manually specified"`
+	Selector string `json:"selector" description:"label keys and values that must match in order to receive traffic for this service; if empty, all pods are selected, if not specified, endpoints must be manually specified"`
```

Comment:
Detail: Since we don't have any non-string-valued labels yet, and even when we do string-valued labels will be the most common by far, I'd omit "string" from the field names (just `value` and `values`).